libs/langgraph/langgraph/graph/state.py

```diff
@@ -949,13 +949,43 @@ def _pick_mapper(
     return partial(_coerce_state, schema)
 
 
-def _coerce_state_pydantic(schema: Type[Any], input: dict[str, Any]) -> dict[str, Any]:
+def _coerce_state_pydantic(
+    schema: Type[BaseModel], input: dict[str, Any], *, __depth__: int = 3
+) -> dict[str, Any]:
+    if __depth__ > 0:
+        for key in input:
+            field = schema.model_fields.get(key)
+            if field is None:
+                continue
+            if (
```

Comment:
what about list[BaseModel] or the like?

Reply:
Probably better than mine but:
 https://github.com/langchain-ai/langgraph/pull/3806/files